atmosphere/model.cc

```diff
@@ -448,11 +447,9 @@ GLuint NewTexture3d(int width, int height, int depth, GLenum format,
   glTexParameteri(GL_TEXTURE_3D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
   glTexParameteri(GL_TEXTURE_3D, GL_TEXTURE_WRAP_R, GL_CLAMP_TO_EDGE);
   glBindBuffer(GL_PIXEL_UNPACK_BUFFER, 0);
-  GLenum internal_format = format == GL_RGBA ?
-      (half_precision ? GL_RGBA16F : GL_RGBA32F) :
-      (half_precision ? GL_RGB16F : GL_RGB32F);
+  GLenum internal_format = half_precision ? GL_RGBA16F : GL_RGBA32F;
   glTexImage3D(GL_TEXTURE_3D, 0, internal_format, width, height, depth, 0,
-      format, GL_FLOAT, NULL);
+      GL_RGBA, GL_FLOAT, NULL);
```

Comment:
I think forcing RGBA is a big drawback here, it increases the GPU memory needs by 33% when this is not always necessary. Could you use 'format' by default and fallback to RGBA only if you get a framebuffer completness error?